nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -132,74 +130,42 @@ protected void doRefresh() {
 
   @Override
   protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    try {
+      Content content =
+          client.getApi().getContent().key(key).reference(client.getReference()).get().get(key);
+      if (content != null) {
```

Comment:
I don't understand the purpose of this check. The commit functionality in Nessie checks for content type mismatches.

Reply:
`createTableViaTransactionThatAlreadyExistsAsView` from `ViewCatalogTests` will fail without this check. It throws `NessieBadRequestException` without this check. 

doRefresh() won't be called when Iceberg calls `commitTransaction` from above testcase. Hence, it needs this check.

Reply:
Still, why is this _always_ run and not only when a conflict happened?

Reply:
Above I got a comment from Alex that we should not catch `NessieBadRequestException` in the client code. Hence, I was doing it for all the cases. 

Now, I am catching `NessieBadRequestException` and checking the content existence only during this error case.